src/pymor/analyticalproblems/expressions.py

```diff
@@ -379,43 +481,95 @@ def _broadcastable_shapes(first, second):
-class abs(UnaryFunctionCall):      numpy_symbol = 'abs'      # NOQA
-class sign(UnaryFunctionCall):     numpy_symbol = 'sign'     # NOQA
+class Sum(BinaryOp):  numpy_symbol = '+'; fenics_op = operator.add      # NOQA
+class Diff(BinaryOp): numpy_symbol = '-'; fenics_op = operator.sub      # NOQA
+class Prod(BinaryOp): numpy_symbol = '*'; fenics_op = operator.mul      # NOQA
+class Div(BinaryOp):  numpy_symbol = '/'; fenics_op = operator.truediv  # NOQA
+
+
+class Pow(BinaryOp):  numpy_symbol = '**'; fenics_op = 'elem_pow'    # NOQA
+class LE(BinaryOp):   numpy_symbol = '<='; fenics_op = 'le'          # NOQA
+class GE(BinaryOp):   numpy_symbol = '>='; fenics_op = 'ge'          # NOQA
+class LT(BinaryOp):   numpy_symbol = '<';  fenics_op = 'lt'          # NOQA
+class GT(BinaryOp):   numpy_symbol = '>';  fenics_op = 'gt'          # NOQA
+class Mod(BinaryOp):  numpy_symbol = '%';  fenics_op = None          # NOQA
+
+
+class sin(UnaryFunctionCall):      numpy_symbol = 'sin';     fenics_op = 'sin'       # NOQA
+class cos(UnaryFunctionCall):      numpy_symbol = 'cos';     fenics_op = 'cos'       # NOQA
+class tan(UnaryFunctionCall):      numpy_symbol = 'tan';     fenics_op = 'tan'       # NOQA
+class arcsin(UnaryFunctionCall):   numpy_symbol = 'arcsin';  fenics_op = 'asin'      # NOQA
+class arccos(UnaryFunctionCall):   numpy_symbol = 'arccos';  fenics_op = 'acos'      # NOQA
+class arctan(UnaryFunctionCall):   numpy_symbol = 'arctan';  fenics_op = 'atan'      # NOQA
+class sinh(UnaryFunctionCall):     numpy_symbol = 'sinh';    fenics_op = 'sinh'      # NOQA
+class cosh(UnaryFunctionCall):     numpy_symbol = 'cosh';    fenics_op = 'cosh'      # NOQA
+class tanh(UnaryFunctionCall):     numpy_symbol = 'tanh';    fenics_op = 'tanh'      # NOQA
+class arcsinh(UnaryFunctionCall):  numpy_symbol = 'arcsinh'; fenics_op = None        # NOQA
+class arccosh(UnaryFunctionCall):  numpy_symbol = 'arccosh'; fenics_op = None        # NOQA
+class arctanh(UnaryFunctionCall):  numpy_symbol = 'arctanh'; fenics_op = None        # NOQA
+class exp(UnaryFunctionCall):      numpy_symbol = 'exp';     fenics_op = 'exp'       # NOQA
+class log(UnaryFunctionCall):      numpy_symbol = 'log';     fenics_op = 'ln'        # NOQA
+class sqrt(UnaryFunctionCall):     numpy_symbol = 'sqrt';    fenics_op = 'sqrt'      # NOQA
+class sign(UnaryFunctionCall):     numpy_symbol = 'sign';    fenics_op = 'sign'      # NOQA
+
+
+class exp2(UnaryFunctionCall):
+    numpy_symbol = 'exp2'
+
+    def fenics_expr(self, params):
+        from ufl import elem_pow
+        f_expr = self.arg.fenics_expr(params)
+        if f_expr.shape:
+            return np.vectorize(lambda x: elem_pow(2, x.item()))(f_expr)
+        else:
+            return np.array(elem_pow(f_expr.item()))
 
 
-class angle(UnaryFunctionCall):
+class log2(UnaryFunctionCall):
+    numpy_symbol = 'log2'
+
+    def log2(x):
+        from ufl import ln
+        return ln(x.item()) / ln(2)
```

Comment:
If you define `log2` here, it will become a method of the class and `x` will act as the `self` argument. You need to move the definition inside of `fenics_expr`. Also, otherwise the name `log2` will not be available inside `fenics_expr`.